src/robothub_oak/watch_dog.py

```diff
@@ -0,0 +1,85 @@
+import logging
+
+from abc import ABC, abstractmethod
+from enum import Enum
+from robothub_core import AGENT
+from robothub_core import RobothubApplication as RhApp
+from threading import Thread
+from time import monotonic
+from typing import Optional
+from uuid import uuid4
+
+__all__ = ["WatchDog", "AgentResponse"]
+
+logger = logging.getLogger(__name__)
+
+
+class AgentResponse(Enum):
+    RESTART = "restart"
+    NOTIFY = "notify"
+
+
+# send_health_check_init(health_check_frequency) -> message = {'what': 'wish', 'type': "health_check", 'body': {"frequency": frequency: float, "action": "restart" or "notify", "id_: "unique uuid", "description": "user defined string"}}
+# message = {'what': 'notification', 'type': "health_check", 'body': {"frequency": frequency: float, "action": "restart" or "notify"}}
+# destroy watch dog: message = {'what': 'wish', 'type': "health_check_destroy", 'body': {"id_: "unique_uuid"}}
+
+
+class WatchDog(ABC):
+    __active_watch_dogs = set()
+
+    @classmethod
+    def create(cls, interval_seconds: float, description: str, agent_response: AgentResponse = AgentResponse.RESTART) -> "WatchDogChild":
```

Comment:
I agree, I just don't see why `create` and `destroy_all` should be static methods. I think it will be useful to hold a ledger `(__active_watch_dogs)` of all created instances in a static variable so that `destroy_all` is possible. In this case, `create` and `destroy_all` need to access the `__active_watch_dogs` variable and for that reason i think they should be class methods.

Reply:
I agree that they should remain class methods because they are used as factory/cleanup functions. But not because they need access to `__active_watch_dogs`, you still can access this variable from static method using `WatchDog.__active_watch_dogs`.